### PR TITLE
feat: add `fn new_dag_cbor_default` to `Cid`

### PR DESCRIPTION
**Pull request by hanabi1224**
## Summary of changes

As part of https://github.com/ChainSafe/forest/issues/2801

Changes introduced in this pull request:

- Add extension method `fn new_dag_cbor_default` to `Cid` to match the default filecoin CID builder in go. See [code](https://github.com/filecoin-project/go-state-types/blob/master/abi/cid.go#L49)

## Reference issue to close (if applicable)

Closes

## Change checklist

- [x] I have performed a self-review of my own code,
- [x] I have made corresponding changes to the documentation,
- [x] I have added tests that prove my fix is effective or that my feature works (if possible),
- [x] I have made sure the [CHANGELOG][1] is up-to-date. All user-facing changes should be reflected in this document.

[1]: https://github.com/ChainSafe/forest/blob/main/CHANGELOG.md

### CODE REVIEW

**Review by lemmih — CHANGES_REQUESTED** [review]
Can you explain a little more about why we want this?

**Comment by hanabi1224** [comment]
> Can you explain a little more about why we want this?

@lemmih In state migration code, sometimes it needs to calculate CID for a cor-serializable object, without calling `CborStore.put_cbor_default` API. One example is calculating CID for DealProposal (see https://github.com/ChainSafe/fil-actor-states/pull/120), I feel it unsustainable to add a method to each struct, it would be much easier to do it with `Cid::new_dag_cbor_default(deal_proposal)`

File: utils/forest_utils/src/cid/mod.rs [hunk 1]
```diff
@@ -0,0 +1,26 @@
...
+// SPDX-License-Identifier: Apache-2.0, MIT
+
+use cid::{
+    multihash::{Code::Blake2b256, MultihashDigest},
+    Cid,
+};
+
+/// Extension methods for constructing `dag-cbor` [Cid]
+pub trait CidCborExt {
+    /// Default CID builder for Filecoin
```
Comment (LesnyRumcajs) [inline utils/forest_utils/src/cid/mod.rs:11]: Such a lovely new trait! It would be a shame not to have 100% coverage on it! :)

**Comment by LesnyRumcajs** [comment]
@hanabi1224, what do we do with this PR?

File: utils/forest_utils/src/cid/mod.rs (hunk 1 above)
Comment (hanabi1224) [reply utils/forest_utils/src/cid/mod.rs:11]: test added

File: utils/forest_utils/src/cid/mod.rs [hunk 2]
```diff
@@ -0,0 +1,47 @@
...
+    ///
+    /// - The default codec is CBOR
```
Comment (lemmih) [inline utils/forest_utils/src/cid/mod.rs:13]: The codec is `DAG_CBOR`, right?

File: utils/forest_utils/src/cid/mod.rs [hunk 3]
```diff
@@ -0,0 +1,47 @@
...
+    /// - The default hash function is 256 bit BLAKE2b
+    ///
+    /// This matches [`abi.CidBuilder`](https://github.com/filecoin-project/go-state-types/blob/master/abi/cid.go#L49) in go
+    fn new_dag_cbor_default<S: serde::ser::Serialize>(obj: &S) -> anyhow::Result<Cid> {
```
Comment (lemmih) [inline utils/forest_utils/src/cid/mod.rs:17]: The naming feels non-intuitive to me. I can see that the `default` part refers to `blake256,` but I don't see any non-default counterpart. How about naming this function `from_cbor` or `from_cbor_blake256`?

**Review by lemmih — CHANGES_REQUESTED** [review]

File: utils/forest_utils/src/cid/mod.rs (hunk 2 above)
Comment (hanabi1224) [reply utils/forest_utils/src/cid/mod.rs]: Fixed.

**Comment by LesnyRumcajs** [comment]
@lemmih @hanabi1224, what do we do with this PR?

**Review by lemmih — APPROVED** [review]

**Comment by lemmih** [comment]
> @lemmih @hanabi1224, what do we do with this PR?

I'm okay with merging it.

**Pull request merged**